### PR TITLE
[Suggestion/Discussion] Support poetry

### PR DESCRIPTION
**Pull request by dannywillems**
Ease the installation and the environment setup for the different scripts.

Example:
```shell
poetry install
poetry run python orchard_poseidon.py
```

See [the poetry website](https://python-poetry.org/) for more information.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+name = "zcash-test-vectors"
+version = "0.1.0"
+description = "Zcash test vectors"
+authors = [
+]
+license = "MIT"
+readme = "README.md"
+homepage = "https://github.com/zcash-hackworks/zcash-test-vectors/"
+repository = "https://github.com/zcash-hackworks/zcash-test-vectors/"
+documentation = "https://github.com/zcash-hackworks/zcash-test-vectors/"
```
Comment (str4d) [inline pyproject.toml:11]: Per https://github.com/python-poetry/poetry/issues/1537#issuecomment-854612263:
```suggestion
documentation = "https://github.com/zcash-hackworks/zcash-test-vectors/"
classifiers = [
    "Private :: Do Not Upload",
]
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[tool.poetry]
+name = "zcash-test-vectors"
+version = "0.1.0"
+description = "Zcash test vectors"
+authors = [
+]
+license = "MIT"
```
Comment (str4d) [inline pyproject.toml:7]:
```suggestion
license = "MIT OR Apache-2.0"
```

**Review by str4d — APPROVED** [review]
Nice, it's like `cargo` for Python!

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[tool.poetry]
+name = "zcash-test-vectors"
+version = "0.1.0"
+description = "Zcash test vectors"
+authors = [
```
Comment (str4d) [inline pyproject.toml:5]:
```suggestion
authors = [
    "Jack Grigg <jack@z.cash>",
    "Daira Hopwood <daira@jacaranda.org>",
    "ying tong <yingtong@z.cash>",
    "Taylor Hornby <taylor@defuse.ca>",
    "Kris Nuttycombe <kris@electriccoin.co>",
    "Simon <simon@bitcartel.com>",
    "Ariel Gabizon <ariel.gabizon@gmail.com>",
    "Deirdre Connolly <durumcrustulum@gmail.com>",
```

**Review by daira — COMMENTED** [review]
Post-hoc ACK

**Pull request merged**